### PR TITLE
feat: Add doc to add a sdk

### PR DESCRIPTION
**Pull request by gfyrag**
# Add doc to create a SDK

Add a documentation to add a new SDK (see ./SDK.md).
Also add a specific PR template.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Technical debt
- [X] Doc

### CODE REVIEW

**Review by flemzord — DISMISSED** [review]

File: SDK.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+The filename MUST match the name of the involved api generator (ie java.yaml for java or php.yaml for php).
+
+By convention, the newly added config file MUST add the following configuration : 
+```
+templateDir: templates/<generator name> # related to sdk/
+```
+
+Again, we use the generator name to name this directory. We also have to create the directory sdk/templates/<generator name>.
+
+If the generator provide required properties, you MUST define the licence to MIT.
```
Comment (jdupas22) [inline SDK.md:22]: i think we should rephrase this, seems unclear to me

Comment (gfyrag) [reply SDK.md:22]: Ok. What point is not clear for you?

File: .github/PULL_REQUEST_TEMPLATE/pull_request_sdk_template.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# SDK : OPEN API GENERATOR NAME
+
+__Please add a description to this PR.__
+
+## How to deploy
+
+__Please provide steps to deploy the generated sdk to official channels.__
+__Please be the more exhaustive you can__
```
Comment (antoinegelloz) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_sdk_template.md]:
```suggestion
__Please be as exhaustive as possible__
```

File: SDK.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# Add a new SDK
+
+We use [OpenAPI generator](https://openapi-generator.tech/) to generate ledger client SDK.`
+This doc refers to the tools named [task](https://taskfile.dev/)
```
Comment (antoinegelloz) [inline SDK.md]:
```suggestion
This doc refers to the tool named [task](https://taskfile.dev/)
```

File: SDK.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+If you want to add a new languages, you have to configure some things.
```
Comment (antoinegelloz) [inline SDK.md]:
```suggestion
If you want to add a new language, you have to configure a few things.
```

File: SDK.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+## Configuration
+
+First, you need to create a configuration file.
+
+The config file must be added inside the directory sdk/configs/.
```
Comment (antoinegelloz) [inline SDK.md]:
```suggestion
This file must be added inside the sdk/configs directory.
```

File: SDK.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+The filename MUST match the name of the involved api generator (ie java.yaml for java or php.yaml for php).
```
Comment (antoinegelloz) [inline SDK.md]:
```suggestion
The file name MUST match the name of the involved api generator (i.e. java.yaml for java or php.yaml for php).
```

File: SDK.md (hunk 1 above)
Comment (antoinegelloz) [reply SDK.md:22]: I don't understand this sentence neither 😢

File: SDK.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+```
+templateDir: templates/<generator name> # related to sdk/
+```
+
+Again, we use the generator name to name this directory. We also have to create the directory sdk/templates/<generator name>.
+
+If the generator provide required properties, you MUST define the licence to MIT.
+
+The rest of the configuration is specific to the language and should be carefully specified. 
+You have to keep in mind the SDK will be hosted on github.com/numary/numary-sdk-<generator name> and so, you could have to define some properties for a language which would not exists on another language.
```
Comment (antoinegelloz) [inline SDK.md]:
```suggestion
You have to keep in mind that the SDK will be hosted on github.com/numary/numary-sdk-<generator name>, so you could have to define some properties for a language which would not exist on another language.
```

File: SDK.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+## Customization
+
+Customization of the generated SDK can be achieved modifying the openapi generator builtins templates.
```
Comment (antoinegelloz) [inline SDK.md]:
```suggestion
Customization of the generated SDK can be achieved by modifying the openapi generator builtins templates.
```

File: SDK.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+You can extract them for your generator using the following command :
+```
+task sdk:template -- <generator name>
+```
+
+The command will extract all templates inside the directory sdk/templates/<generator name>.
+You can modify them as well as you want.
```
Comment (antoinegelloz) [inline SDK.md]:
```suggestion
You can edit them as much as you want.
```

**Review by antoinegelloz — CHANGES_REQUESTED** [review]

File: SDK.md (hunk 1 above)
Comment (gfyrag) [reply SDK.md:22]: Some generator (java for example) provide configuration options to define the licence name. But it is not the case for all.
I want to express the fact, that if a generator provide those configuration options, we have to use them.

Comment (jdupas22) [reply SDK.md:22]: I still don't see it, why would the licence be set to MIT if the generator provide licence configuration ?

Comment (gfyrag) [reply SDK.md:22]: It is just a best effort to set a licence on the generated sdk.

Comment (gfyrag) [reply SDK.md:22]: (MIT was decided with @altitude and @flemzord )

**Review by flemzord — APPROVED** [review]

**Pull request merged**